Spanner/src/Result.php

```diff
@@ -136,14 +141,16 @@ public function __construct(
         callable $call,
         $transactionContext,
         ValueMapper $mapper,
-        $retries = 3
+        $retries = 3,
+        &$transactionId = null
```

Comment:
Why is this passed by reference? There is no reason to pass this by reference here.

If this value becomes updated at some point, and the user needs the value, the way this should be handled is by adding a `getTransactionId` method on `Result`, not by passing by reference.

Reply:
This transactionId is an attribute of the Transaction and is not accessible here. Either we need to pass the Transaction object here or this variable reference.

Reply:
I cannot speak to the necessity of passing in the `Transaction` object, and whether that's the best design or not. I can say that the design here, passing in the variable by reference, is not acceptable.

Without diving deeper, I agree that having a `Transaction` object on the `Result`, and allowing the user to call `getTransaction`, sounds like a more acceptable pattern.